src/lib.rs

```diff
@@ -1003,13 +1035,19 @@ impl<'a, 'input: 'a> Node<'a, 'input> {
     /// assert_eq!(doc.root_element().namespaces().len(), 1);
     /// ```
     #[inline]
-    pub fn namespaces(&self) -> &'a [Namespace<'input>] {
-        match self.d.kind {
+    pub fn namespaces(&self) -> impl ExactSizeIterator<Item = &'a Namespace<'input>> + fmt::Debug {
```

Comment:
I we do this, this should become a named type as having unnameable types in the API of a library is a discourtesy to its users.

Reply:
We will be able to save a ton of code when https://github.com/rust-lang/rust/issues/63063 gets stabilized and comes within the MSRV reach of this crate. Until then, I added an explicit type.

I also did some boy scouting on `Descendants` to remove some redundant bounds check which I guess shows in the benchmarks below.